src/kakarot/execution_context.cairo

```diff
@@ -234,6 +138,25 @@ namespace ExecutionContext {
         );
     }
 
+    // @notice Finalizes the execution context.
```

Comment:
is it worth linking here to some sort of documentation about the 'rules' of using dictionaries (and indirectly how memory and the stack are implemented as dictionaries) and how finalizing for the memory and stack involves squashing? 

https://www.cairo-lang.org/docs/reference/common_library.html#common-library-default-dict

From Eni's blogpost about sierra

>  Every Dict must be squashed at the end of a program by calling the dict_squash(Dict<K,V>) -> () function, which verifies the consistency of the sequence of key updates. Unsquashed dictionaries are dangerous, as a malicious prover could prove the correctness of inconsistent updates.

https://medium.com/nethermind-eth/under-the-hood-of-cairo-1-0-exploring-sierra-7f32808421f5